infrastructure-configs/demeter.json

```diff
@@ -3,14 +3,37 @@
 "substitutions": {
     "AOCS_HDFS_BAMS": "hdfs:///datasets/ovarian/ega-tim",
     "HG19_FASTA": "/hpc/users/ahujaa01/reference_genomes/hg19-reference/ucsc.hg19.fasta"
-}
+},
 
-"spark_configuration": {
-    "spark.driver.maxResultSize": "0",
-    "spark.yarn.executor.memoryOverhead": "6000",
-    "spark.executor.memory": "16g",
-    "spark.default.parallelism": "10000"
+"spark_submit_arguments": [
+    "--deploy-mode", "cluster",
+    "--master", "yarn",
+    "--executor-cores", "6",
+    "--executor-memory", "17g",
+    "--driver-memory", "10g",
 
-}
+    "--conf", "spark.driver.maxResultSize=0",
```

Comment:
It's just a question of whether we should have a bunch of strings here in this JSON file, or put them in their own file with a cleaner syntax, no `"--conf"` repetition, and better escaping/grouping guarantees ([Spark docs](http://spark.apache.org/docs/latest/configuration.html#dynamically-loading-spark-properties)). 

In both cases if we want to change a default value for one of these confs, we make the change to a git-tracked file, either this `demeter.json` or the `conf` file; no difference there.

Are you sure `"--conf", "spark.executor.extraJavaOptions=-XX:+PrintGCDetails -XX:+PrintGCTimeStamps -verbose:gc"` is going to get parsed properly?

You can always pass args on the cmdline which will override whatever's in a config file, or indeed any args that were previously specified on the cmdline; also no difference there.